merge_sort/Java/MergeSort.java

```diff
@@ -0,0 +1,63 @@
+import java.util.Random;
+import java.util.Scanner;
+
+//Sorts an array, of given size, of randomly generated numbers using MergeSort Algorithm
+public class MergeSort {
+
+    //Recursively call mergesort, reducing the array size in every call
+    void mergesort(int []arr,int l,int h){
+        int mid = (l+h)/2;		//Split array into two virtual parts
+        if(h>l){			//If array contains more than two elements
+            mergesort(arr,l,mid);	//Sort first part recursively
+            mergesort(arr,mid+1,h);	//Sort second part recursively
+            merge(arr,l,mid,h);		//Merge sorted arrays into single sorted array
```

Comment:
Line contains following spacing inconsistencies:
- Tabs used instead of spaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpqragilip/merge_sort/Java/MergeSort.java
+++ b/tmp/tmpqragilip/merge_sort/Java/MergeSort.java
@@ -16,7 +16,7 @@
 
     //Merge two sorted arrays into a single sorted array
     private void merge(int []arr,int l,int m,int h){
-        int []c = new int[l+h];		//create temporary array. This will hold the sorted array
+        int []c = new int[l+h];     //create temporary array. This will hold the sorted array
 
     //Copies of l,m,h. These will be modified, while originals remain unchanged.
         int i=l;
```